confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java

```diff
@@ -83,9 +86,38 @@ public int getDeviceGroupID(String device) {
     return hashExecutor.getDeviceGroupID(device);
   }
 
-  // TODO: Interfaces for metadata operations
+  /** Build ConfigNodeGroup ConsensusLayer */
+  private void setConsensusLayer(ConfigNodeConf config) {
+    // TODO: Support other consensus protocol
+    this.consensusImpl =
+        new StandAloneConsensus(
+            id -> {
+              if (id.getType() == GroupType.PartitionRegion) {
+                return new StandAloneStateMachine();
+              } else {
+                return new EmptyStateMachine();
+              }
+            });
+    this.consensusImpl.start();
+
+    this.consensusGroupId = new ConsensusGroupId(GroupType.PartitionRegion, 0);
+    this.consensusImpl.addConsensusGroup(
+        this.consensusGroupId,
+        Collections.singletonList(
+            new Peer(
```

Comment:
We may need to support construct PartitionRegionConsensusGroup from conf here?

Reply:
I think optimize this code in next PR would be better. So that I can discuss with Ziyang